### ISSUE TITLE
Multiple calls to modal.open() from parent element stacking backdrops

### ISSUE THREAD

**Issue by kcrossman** [dougludlow/ng2-bs3-modal#149]
I have a modal where its open() method is being called multiple times from Angular2. During each call to open(), a backdrop is created behind the modal while the modal closes and re-opens itself. Clicking on anything that closes the modal only removes the last backdrop. For example, calling open() manually 3 times creates 3 backdrops and closing the modal only closes the last one, leaving 2 backdrops still remaining...

Temporary solution: My own show() method deletes all backdrops before calling open().

The ideal solution should be to prevent this behavior. If I were to write my own modal component, for example, there would only be 1 backdrop in the most-parent component, so it would not be possible to have multiple backdrops. However, I realize since this is a bit more modular stuff like this can happen...

I am currently using version: "0.10.4".

Do other people have this issue?

```
import { Component, ViewChild } from '@angular/core';
import { ModalComponent } from 'ng2-bs3-modal/ng2-bs3-modal';

@Component({
    selector: 'test-modal',
    template: ...
})
export class TestModal {
    @ViewChild(ModalComponent) modal: ModalComponent;

    show() {
        this.removeElementsByClass('modal-backdrop');
        this.modal.open();
    }

    private removeElementsByClass(className) {
        let elements = document.getElementsByClassName(className);
        while (elements.length > 0) {
            elements[0].parentNode.removeChild(elements[0]);
        }
    }
}
```

**Comment by dougludlow** [comment]
@kcrossman wouldn't the ideal behavior be that if open is called and the modal is already open, that nothing happens?

**Comment by kcrossman** [comment]
Preventing open based on if modal.visible is true only works if the modal has appeared. In my case, it would be called multiple times before it had actually opened so the backdrops duplicate before the modal is done opening. This did work for a time but using the modal in many different components proved that it wasn't sufficient at removing the backdrops. Also, the most-recent version of the modal is not open, correct (potentially)? That could cause other problems with possibly bad data.

**Comment by dougludlow** [comment]
No, this would probably need to be an internal change. I'd need to track the state of whether or not it is opening and cancel any calls to `open()` if the modal is opening or open. I'm a little busy at the moment, but I'll see what I can cook up here when I find the time.

**Comment by kcrossman** [comment]
Wouldn't you want to preserve the most-recent open instead?

**Comment by kevado** [comment]
I have the same issue but I in my case the body "modal-open" class does not match the "modal-backdrop fade in" class. Meaning if I open and close modals within a certain timeframe of each other there seems to be some confusion manipulating the DOM thus leaving the two states out of sync. The end result show I ended up with one "modal-backup fade in" but no "modal-open" on the body.

**Comment by dougludlow** [comment]
This should be fixed in v0.11
